### PR TITLE
Give permission to users for uploadIso

### PR DESCRIPTION
**Pull request by bharkarampudi**
### Description

This PR...

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

#### Feature/Enhancement Scale

- [ ] Major
- [ ] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [ ] Minor
- [ ] Trivial

### CODE REVIEW

File: api/src/org/apache/cloudstack/api/command/user/iso/GetUploadParamsForIsoCmd.java [hunk 1]
```diff
@@ -32,12 +32,14 @@
...
 import org.apache.cloudstack.api.response.ZoneResponse;
 import org.apache.cloudstack.context.CallContext;
+import org.apache.cloudstack.acl.RoleType;
 
 import java.net.MalformedURLException;
 
 @APICommand(name = GetUploadParamsForIsoCmd.APINAME,
         description = "upload an existing ISO into the CloudStack cloud.",
         responseObject = GetUploadParamsResponse.class, since = "4.13",
+        authorized = {RoleType.Admin, RoleType.ResourceAdmin, RoleType.DomainAdmin, RoleType.User},
```
Comment (nlgordon) [inline api/src/org/apache/cloudstack/api/command/user/iso/GetUploadParamsForIsoCmd.java:42]: GetUploadParamsForVolumeCmd doesn't have this, but users without root admin role can still upload volumes. Do we know why this is required for this API?

**Review by nlgordon — DISMISSED** [review]

File: api/src/org/apache/cloudstack/api/command/user/iso/GetUploadParamsForIsoCmd.java (hunk 1 above)
Comment (bharkarampudi) [reply api/src/org/apache/cloudstack/api/command/user/iso/GetUploadParamsForIsoCmd.java:42]: @nlgordon For GetUploadParamsForVolumeCmd, permissions are given as database entries in create-default-role-api-mappings.sql file. we can give permissions in two ways one is using annotations and second is using role permission tables

Comment (Slair1) [reply api/src/org/apache/cloudstack/api/command/user/iso/GetUploadParamsForIsoCmd.java:42]: @bharkarampudi @nlgordon to be consistent i would say it is best to do it via role permissions.  does the community version update that sql file?

Comment (nlgordon) [reply api/src/org/apache/cloudstack/api/command/user/iso/GetUploadParamsForIsoCmd.java:42]: I looked at main, and it does not explicitly grant access to this API command by default.

Comment (bharkarampudi) [reply api/src/org/apache/cloudstack/api/command/user/iso/GetUploadParamsForIsoCmd.java:42]: In main, the access is granted through command like we did.

File: api/src/org/apache/cloudstack/api/command/user/iso/GetUploadParamsForIsoCmd.java [hunk 2]
```diff
@@ -32,12 +32,14 @@
 import org.apache.cloudstack.api.response.GuestOSResponse;
 import org.apache.cloudstack.api.response.ZoneResponse;
 import org.apache.cloudstack.context.CallContext;
+import org.apache.cloudstack.acl.RoleType;
 
 import java.net.MalformedURLException;
 
 @APICommand(name = GetUploadParamsForIsoCmd.APINAME,
         description = "upload an existing ISO into the CloudStack cloud.",
         responseObject = GetUploadParamsResponse.class, since = "4.13",
```
Comment (Slair1) [inline api/src/org/apache/cloudstack/api/command/user/iso/GetUploadParamsForIsoCmd.java]: @bharkarampudi  it says 4.13 here, should it say 4.11?

Comment (bharkarampudi) [reply api/src/org/apache/cloudstack/api/command/user/iso/GetUploadParamsForIsoCmd.java]: @Slair1 Yes, it should be 4.11 according to our version, I think it is ported from future version from main so it is 4.13. Shall I change it to 4.11?

Comment (idevalois-ipp) [reply api/src/org/apache/cloudstack/api/command/user/iso/GetUploadParamsForIsoCmd.java]: @Slair1 Confirm you'd like 4.13 changed to 4.11 here?

Comment (idevalois-ipp) [reply api/src/org/apache/cloudstack/api/command/user/iso/GetUploadParamsForIsoCmd.java]: @bharkarampudi Sean has confirmed this should be 4.11.

Comment (bharkarampudi) [reply api/src/org/apache/cloudstack/api/command/user/iso/GetUploadParamsForIsoCmd.java]: Done changes

**Review by Slair1 — APPROVED** [review]
Checked Main, authorized is done the same there

**Pull request merged**